app/src/modules/battery/battery.c

```diff
@@ -51,66 +46,38 @@ static int charger_read_sensors(float *voltage, float *current, float *temp)
 	sensor_channel_get(charger, SENSOR_CHAN_GAUGE_AVG_CURRENT, &value);
 	*current = (float)value.val1 + ((float)value.val2 / 1000000);
 
+	sensor_channel_get(charger, SENSOR_CHAN_NPM1300_CHARGER_STATUS, &value);
+	*chg_status = value.val1;
+
 	return 0;
 }
 
 static void sample(void)
 {
-	/* Fuel gauge variables */
+	int err;
+	int chg_status;
+	bool cc_charging;
 	float voltage;
 	float current;
 	float temp;
 	float state_of_charge;
 	float delta;
-	float time_to_empty;
-	float time_to_full;
-	static struct nrf_fuel_gauge_state_info fg_state;
-	int ret;
-
-	/* Update fuel gauge state */
-	ret = charger_read_sensors(&voltage, &current, &temp);
-	if (ret) {
-		LOG_ERR("Error: Could not read from charger device: %d", ret);
+
+	err = charger_read_sensors(&voltage, &current, &temp, &chg_status);
+	if (err) {
+		LOG_ERR("charger_read_sensors, error: %d", err);
 		SEND_FATAL_ERROR();
 		return;
 	}
 
 	delta = (float) k_uptime_delta(&fg_ref_time) / 1000.f;
-	fg_ref_time = k_uptime_get();
-	state_of_charge = nrf_fuel_gauge_process(voltage, current, temp, delta, &fg_state);
-
-	/* calculate TTE/TTF */
-	time_to_empty = nrf_fuel_gauge_tte_get();
-	time_to_full = nrf_fuel_gauge_ttf_get((bool)-max_charge_current, -term_charge_current);
 
-	fg_burst_current_accumulated += delta * current;
-	fg_burst_delta_accumulated += delta;
-
-	/* Repeat readings upto FG_BURST_COUNT measurement until we collect enough data points */
-	if (fg_burst_index++ < FG_BURST_COUNT) {
-		k_sleep(FG_BURST_DELAY);
-		sample();
-		return;
-	}
+	cc_charging = ((chg_status & NPM1300_CHG_STATUS_CC_MASK) != 0);
 
-	current = fg_burst_current_accumulated / fg_burst_delta_accumulated;
+	state_of_charge = nrf_fuel_gauge_process(voltage, current, temp, delta, NULL);
 
 	LOG_DBG("State of charge: %f", (double)roundf(state_of_charge));
-	LOG_DBG("Voltage: %f", (double)roundf(1000 * voltage));
-	LOG_DBG("Current: %f", (double)roundf(1000 * current));
-
-	if (isnormal(time_to_full)) {
-		LOG_DBG("Time to full: %f", (double)roundf(time_to_full));
-	}
-
-	if (isnormal(time_to_empty)) {
-		LOG_DBG("Time to empty: %f", (double)roundf(time_to_empty));
-	}
-
-	/* Reset burst variables */
-	fg_burst_index = 0;
-	fg_burst_current_accumulated = 0;
-	fg_burst_delta_accumulated = 0;
+	LOG_DBG("The battery is %s", cc_charging ? "charging" : "not charging");
```

Comment:
`cc_charging` is only one of the possible charging states. If you want to split into `charging` and `not charging` you have to look at `chg_status` and test these 3 bits:
![image](https://github.com/hello-nrfcloud/firmware/assets/6489829/35e14b15-0a39-4157-973e-646718f4eff8)

Reply:
You can also look at the current sign btw. That's probably the easiest. We define negative current as charging:
`LOG_DBG("The battery is %s", signbit(current) ? "charging" : "not charging");`